File: backend/src/main/java/br/com/calciolari/datahub/imports/domain/hints/IncompleteDateRange.java
package br.com.calciolari.datahub.imports.domain.hints;

import java.util.Objects;

/**
 * Inclusive period hint from a filename such as {@code 01_07-20_07}.
 * Provenance: {@code INFERRED_DATA} only.
 */
public record IncompleteDateRange(IncompleteDate start, IncompleteDate end) {

	public IncompleteDateRange {
		Objects.requireNonNull(start, "start");
		Objects.requireNonNull(end, "end");
	}
}

File: backend/src/Calciolari.DataHub/Imports/Domain/Hints/IncompleteDateRange.cs
namespace Calciolari.DataHub.Imports.Domain.Hints;

/// <summary>
/// Inclusive period hint from a filename such as 01_07-20_07.
/// Provenance: INFERRED_DATA only.
/// </summary>
public sealed record IncompleteDateRange(IncompleteDate Start, IncompleteDate End);
